core/client/src/main/java/com/thousandeyes/sdk/pagination/Paginator.java

```diff
@@ -0,0 +1,131 @@
+package com.thousandeyes.sdk.pagination;
+
+import java.lang.reflect.InvocationTargetException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+
+import com.thousandeyes.sdk.client.ApiException;
+
+
+
+public class Paginator<T, R> implements Iterable<T> {
+    private static final Pattern CURSOR_PATTERN = Pattern.compile("cursor=([^&]*)");
+    private final Function<String, R> apiCall;
+    private final Function<R, List<T>> dataExtractor;
+
+    public Paginator(Function<String, R> apiCall, Function<R, List<T>> dataExtractor)
+    {
+        this.apiCall = apiCall;
+        this.dataExtractor = dataExtractor;
+    }
+
+    public List<T> getAll() throws ApiException {
```

Comment:
something for discussing i'm thinking that we might **not** want to make this operation available because it can generate a lot of requests 🤔

Reply:
Yeah, definitely something to consider

Reply:
IMO we better no have a get all option